docs/source/index.rst

```diff
@@ -0,0 +1,275 @@
+Notice that the template extends a base.html. We are going to use application's base template.
+
+Let's create a blueprint in our plugin::
+
+    plugin_blueprint = PluginBlueprint('example', __name__)
+
+In order to create a blueprint of a plugin we need to create PluginBlueprint instance.
+
+Then we can create a function rendering the template::
+
+    @plugin_blueprint.route('/plugin')
+    def hello_plugin():
+    return render_template('example_plugin:index.html', plugin=current_plugin)
+
+We can render any plugin template inside our application specifying name of the plugin before template name in ``render_template`` function, like above.
+We are also passing a current_plugin object to the template.
+
+
+Now in the plugin class we can create a method returning plugin's blueprint::
+
+    def get_blueprint(self):
+        return plugin_blueprint
+
+So our plugin code should look like this::
+
+    from flask import render_template
+    from flask_pluginengine import Plugin, PluginBlueprint, current_plugin
+
+    plugin_blueprint = PluginBlueprint('example', __name__)
+
+
+    @plugin_blueprint.route('/plugin')
+    def hello_plugin():
+        return render_template('example_plugin:index.html', plugin=current_plugin)
+
+
+    class ExamplePlugin(Plugin):
+        """ExamplePlugin
+
+        Just an example plugin
+        """
+
+        def get_blueprint(self):
+            return plugin_blueprint
+
+
+To get more about information see `Plugin`_ API reference.
+
+Minimal Application set up
+--------------------------
+
+To create Flask application using Flask-PluginEngine we need to define it as PluginFlask application::
+
+    from flask_pluginengine import PluginFlask
+    app = PluginFlask(__name__)
+
+    @app.route("/")
+    def hello():
+        return "Hello world!"
+
+    if __name__ == "__main__":
+        app.run()
+
+Now we need to set up the configuration: namespace of plugins we will be using with the application and plugins we want to use.
+Let's use the ``example_plugin`` we created before. For example::
+
+    app.config['PLUGINENGINE_NAMESPACE'] = 'example'
+    app.config['PLUGINENGINE_PLUGINS'] = ['example_plugin']
+
+Next we need to create an instance of PluginEngine for our app and load the plugins::
+
+    plugin_engine = PluginEngine(app=app)
+    plugin_engine.load_plugins(app)
+
+Then we can access loaded plugins by calling :func:`get_active_plugins` method, it will return a dictionary.
+If we want to check if all of the plugins were loaded correclty we can also call :func:`get_failed_plugins` method that will return a dictionary with plugins that failed to load::
+
+    active_plugins = plugin_engine.get_active_plugins(app=app).values()
+
+
+Example application functionality
+---------------------------------
+
+To understand the possibilities of Flask-PluginEngine we will create a Jinja template, where we will list all the plugins we are using.
+Let's create two templates, the base one (base.html), for example::
```

Comment:
I'd reword it a bit to:
"Let's create two templates, `base.html` and `index.html`: " and add comments above each code block below, like this:
```jinja
{# base.html #}

<!DOCTYPE html>
 
<html lang="en">
    <head>
        ...
```
```jinja
{# index.html #}

<!DOCTYPE html>
 
<html lang="en">
    <head>
        ...
```